src/components/PostDetails.tsx

```diff
@@ -1,106 +1,115 @@
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+        {!isCommentLoading && !hasErrorMessage && (
+          <>
+            {!hasComments && (
+              <p className="title is-4" data-cy="NoCommentsMessage">
+                No comments yet
+              </p>
+            )}
 
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
-            <div className="message-body" data-cy="CommentBody">
-              One more comment
-            </div>
-          </article>
+            {hasComments && (
+              <>
+                <p className="title is-4">Comments:</p>
+
+                {comments?.map(comment => {
+                  const { id, name, email, body } = comment;
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+                  return (
+                    <article
+                      className="message is-small"
+                      data-cy="Comment"
+                      key={id}
+                    >
+                      <div className="message-header">
+                        <a href={`mailto:${email}`} data-cy="CommentAuthor">
+                          {name}
+                        </a>
+                        <button
+                          data-cy="CommentDelete"
+                          type="button"
+                          className="delete is-small"
+                          aria-label="delete"
+                          onClick={() => handleDeleteComment(id)}
+                        >
+                          delete button
+                        </button>
+                      </div>
 
+                      <div className="message-body" data-cy="CommentBody">
+                        {body}
+                      </div>
+                    </article>
+                  );
+                })}
+              </>
+            )}
+
+            {isFormOpen ? (
+              <NewCommentForm
+                setHasErrorCreateComment={setHasErrorCreateComment}
+              />
+            ) : (
               <button
-                data-cy="CommentDelete"
+                data-cy="WriteCommentButton"
                 type="button"
-                className="delete is-small"
-                aria-label="delete"
+                className="button is-link"
+                onClick={() => {
+                  setIsFormOpen(true);
+                  setHasErrorCreateComment(false);
+                }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here